### DISCUSSION TITLE
How to compile SUMO & JuPedSim

### DISCUSSION THREAD

**Discussion by Ozaq** [PedestrianDynamics/jupedsim#1302] [Q&A]
Moving this question from @pzambare3 to here:

Original Question below:

> Hi
> I am trying to integrate SUMO and JuPedSim .For that I followed the following link, :https://sumo.dlr.de/docs/Installing/Linux_Build.html#how_to_build_jupedsim_and_then_build_sumo_with_jupedsim
> but I got the error with cloning. can you tell me where and how to access sumo and JuPedSim integration module.
> 
> Thank You.

**Comment by Ozaq** [comment]
Which repository did you try to clone? And which error did you get?

The URL to JuPedSim looks fine and can be cloned without issues. @behrisch I assume the doc is up to date?

**Comment by behrisch** [comment]
Yes, I just retried and `git clone https://github.com/PedestrianDynamics/jupedsim.git` works as documented and expected

**Comment by pzambare3** [comment]
@Ozaq and @behrisch Thank you for quick response!

I have cloned jupedsim successfully, simulated example1.py and can see output on console.
So, sumo and jupedsim are working fine separately. for Integration I follow the this link:[ sumo/docs/web/docs/Installing/Linux_Build.md at main · eclipse-s...](https://teams.microsoft.com/l/message/19:2f2e7da00a594e04a39c383432d3916c@thread.v2/1701911593061?context=%7B%22contextType%22%3A%22chat%22%7D)
But I fail to execute this command:  python %SUMO_HOME%/tools/import/dxf/dxf2jupedsim.py polygons.dxf.
Main issue with extraction of .dxf file. Can you please tell me how to deal with this. 

Thank You!

**Reply by behrisch** [reply]
The link needs MS Teams which I do not have here but I suppose you are referring to this doc:  https://sumo.dlr.de/docs/Simulation/Pedestrians.html#jupedsim. The dxf part is completely optional. Please start without it and report your findings.

**Comment by pzambare3** [comment]
Can you please provide proper step-by-step guide to integrate JuPedsim with Sumo?

**Reply by Ozaq** [reply]
The link you provided contains the step-by-guide. Without additional information, i.e. errors or console output I am unable to help.

**Comment by AligetiDivya** [comment]
Hi sir,
I have built jupedsim after which I built sumo as said in the document (sumo-jupedsim).
I have created a pedestrian only scenario manually in Netedit, I have added the below line in input section of .sumo.cfg after net-file and route-file.
<pedestrian.model value="jupedsim"/>

######### Below is the error I'm getting when .sumo.cfg is loaded in sumo-gui

#################

Warning: Environment variable SUMO_HOME is not set properly, disabling XML validation. Set 'auto' or 'always' for web lookups.
Loading net-file from '/home/divya/sumo/Simu.net.xml' ... done (15ms).
Loading done.
Error: Unknown pedestrian model 'jupedsim'
Quitting (on error).
#############################################################################

Is there something that I'm missing? Help me in this regard!

**Reply by behrisch** [reply]
Can you post the output of your initial cmake call when preparing the SUMO build? It should mention whether it actually found JuPedSim.

**Comment by AligetiDivya** [comment]
-- Found osg: /usr/lib/x86_64-linux-gnu/libosg.so  
-- Found OpenThreads: /usr/lib/x86_64-linux-gnu/libOpenThreads.so  
-- Found OpenSceneGraph: /usr/lib/x86_64-linux-gnu/libosgGA.so;/usr/lib/x86_64-linux-gnu/libosgViewer.so;/usr/lib/x86_64-linux-gnu/libosgUtil.so;/usr/lib/x86_64-linux-gnu/libosgDB.so;/usr/lib/x86_64-linux-gnu/libosgText.so;/usr/lib/x86_64-linux-gnu/libosg.so;/usr/lib/x86_64-linux-gnu/libOpenThreads.so (found suitable version "3.6.5", minimum required is "3.4.0") 
**-- Could NOT find JuPedSim! Skipping JuPedSim integration.**
-- Found Gettext: /usr/bin/msgmerge (found version "0.21") 
-- Found Java: /usr/bin/java (found version "11.0.21") found components: Development 

Hello sir, yeah, it says could not find jupedsim!

**Comment by AligetiDivya** [comment]
Hello sir, Can you suggest any method for the sumo build to recognize the jupedsim.
I did everything exactly from the sumo documentation but not working out and i cant find where im making the mistake.
thanks in advance!

**Comment by pzambare3** [comment]
@behrisch and @AligetiDivya : I am also getting same error.
-- Could NOT find JuPedSim! Skipping JuPedSim integration.
-- Found Gettext: /usr/bin/msgmerge (found version "0.21")
-- Found Java: /usr/bin/java (found version "11.0.21") found components: Development

**Comment by behrisch** [comment]
Can you please check that your jupedsim-install dir is properly created and is located right next to your sumo dir? Also pleaee check that your jupedsim-install dir has two subdirs lib and include where the first includes the libjupedsim.so.
So the layout should be something like this:
- jupedsim
- jupedsim-install
  - include
  - lib
- sumo

**Comment by AligetiDivya** [comment]
Hello @behrisch  / @pzambare3,

when i clone the jupedsim, i can get only jupedsim and jupedsim-build directories in my local.
So, i  have created this directory **jupedsim-install.**

**cmake -B jupedsim-build -DCMAKE_INSTALL_PREFIX=../jupedsim-install jupedsim
cmake --build jupedsim-build**

Above 2 commands are working fine but when i hit the below command, the following error persists. Help me in this regard!

**cmake --install jupedsim-build**

cmake error at jupedsim-build/libjupedsim/cmake_install.cmake:52 (file): file install cannot set permissions on "/home/jupedsim-install/lib/libjupedsim.so": operation not permitted. call stack (most recent call first): jupedsim-build/cmake_install.cmake:62 (include)

**Comment by behrisch** [comment]
I am very sorry. I just rechecked the instructions and there was indeed a typo. The correct first command  is 
```
cmake -B jupedsim-build -DCMAKE_INSTALL_PREFIX=jupedsim-install jupedsim
```
I updated the docs already.

**Comment by pzambare3** [comment]
@behrisch : Thank you very much for the update.

**Comment by AligetiDivya** [comment]
@behrisch 
Hello sir,

Finally Sumo built found jupedsim.
**-- Found JuPedSim: /home/divya/jupedsim-install/lib/cmake/jupedsim 1.0.5**

But when i add the below line in the .sumo.cfg file,
**<pedestrian.model value="jupedsim"/>**
And when i load the configuration to sumo-gui, it says,
**----------------------------------------------------------------------------------------
Loading net-file from '/home/divya/sumo/tools/S_scenario/Simu.net.xml' ... done (10ms).
Loading done.
Error: Unknown pedestrian model 'jupedsim'
Quitting (on error).**

Is there anything am I missing, kindly let me know?
Additional info: I have created my .net.xml manually using Netedit

**Comment by behrisch** [comment]
Please make sure you are using the sumo-gui you built yourself. Have a look at the about dialog (from the help menu) which version it is and whether it really has been compiled with jupedsim

**Comment by AligetiDivya** [comment]
hello sir @behrisch,
![screen](https://github.com/PedestrianDynamics/jupedsim/assets/37445617/832dc839-7e31-41f7-91df-c792f897499b)
i can see jupedsim option enabled in my sumo1.19
when i load a .sumocfg file without the below line
<pedestrian.model value="jupedsim"/>
I can run the simulation.
is there anything i'm missing ?

**Reply by bcoueraud87** [reply]
As mentionned by my colleague above, have a look at the about dialog, you should see JuPedSim written:

![image](https://github.com/PedestrianDynamics/jupedsim/assets/56519579/e0be1cd6-9ac8-44d4-9c12-faa9a3417cb6)

That means that your SUMO was built with JuPedSim (either with the release version of JuPedSim that comes with the SUMO distribution, or the one that you built yourself if you did so). Now, to use JuPedSim as a pedestrian model, you can add `<pedestrian.model value="jupedsim"/>` to your config file  as you know, note that you could also use the command line argument `--pedestrian.model jupedsim` as an alternative. If you want now to be sure that this has been taken into account, then when you run your scenario, you will see that the geometry used by JuPedSim (that corresponds to the pedestrian lanes) are drawn on the screen in cyan (as a default) and you will have a message in the SUMO console that will tell you that this geometry has been correctly generated for JuPedSim (or not, if you happen to have an issue with the geometry). Hope this helps.
